Examples/SnapinsSDKKotlinExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/activities/MainActivity.java

```diff
@@ -0,0 +1,269 @@
+import com.salesforce.snapinssdkexample.utils.ServiceSDKUtils;
+
+import java.util.Map;
+
+/**
+ * Main test activity supporting basic functionality:
+ * <ul>
+ * <li>Launching Knowledge UI {@link SupportHomeViewAddition}</li>
+ * <li>Authentication (Logging in or out)</li>
+ * </ul>
+ */
+public class MainActivity extends AppCompatActivity implements SosAvailability.Listener {
+    private KnowledgeUI mKnowledgeUI = null;
+    private KnowledgeUIClient mKnowledgeUIClient = null;
+    private TextView knowledgeLaunchButton;
+    private Button loginButton;
+    private Button logoutButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+
+        knowledgeLaunchButton = findViewById(R.id.knowledge_launch_button);
+        loginButton = findViewById(R.id.login_button);
+        logoutButton = findViewById(R.id.logout_button);
+
+        setSupportActionBar(toolbar);
+
+        initializeServiceSDK();
+
+        setupButtons();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+
+        // Clean up SOS
+        if (SosAvailability.isPolling()) {
+            SosAvailability.stopPolling();
+            SosAvailability.removeListener(this);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()){
+            case R.id.action_sos_settings:
+                return startActivityFor(SosSettingsActivity.class);
+            case R.id.action_kb_settings:
+                return startActivityFor(KnowledgeSettingsActivity.class);
+            case R.id.action_chat_settings:
+                return startActivityFor(ChatSettingsActivity.class);
+            case R.id.action_case_settings:
+                return startActivityFor(CaseSettingsActivity.class);
+            case R.id.action_version_page:
+                return startActivityFor(VersionActivity.class);
+            case R.id.action_check_chat_agent_availability:
+                return showLiveAgentChatAvailability();
+            case R.id.action_check_sos_agent_availability:
+                return showSosAgentAvailability();
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    /**
+     * Called when the SOS availability status changes
+     */
+    @Override
+    public void onSosAvailabilityChange(SosAvailability.Status status) {
+        Toast.makeText(this,
+                getString(R.string.sos_agent_availability_change_message) + status.name(),
+                Toast.LENGTH_SHORT).show();
+    }
+
+    public Boolean startActivityFor(Class<? extends AppCompatActivity> activityClass) {
```

Comment:
Non primitive `Boolean`(s). See comment above in `SupportHomeViewAddition.java`

Reply:
done